packages/ui-kit/src/components/multi-code-block.js

```diff
@@ -0,0 +1,192 @@
+`;
+const LanguagesDropDown = styled.select`
+  display: block;
+  font-size: ${typography.fontSizes.body};
+  line-height: ${typography.lineHeights.body};
+  color: ${colors.light.surfacePrimary};
+  padding-right: ${dimensions.spacings.m};
+  box-sizing: border-box;
+  cursor: pointer;
+  border: none;
+  appearance: none;
+  background: transparent;
+
+  option {
+    color: black;
+  }
+
+  ::-ms-expand {
+    display: none;
+  }
+
+  :hover {
+    color: ${colors.light.surfaceInlineCode};
+  }
+
+  :focus {
+    outline: none;
+  }
+`;
+
+const languageDisplayNames = {
+  sh: 'Terminal',
+  zsh: 'Terminal',
+  console: 'Terminal',
+  bash: 'Terminal',
+  js: 'JavaScript',
+  javascript: 'JavaScript',
+  java: 'Java',
+  php: 'PHP',
+  cs: 'C#',
+  ts: 'TypeScript',
+  python: 'Python',
+  go: 'Go',
+  swift: 'Swift',
+  ruby: 'Ruby',
+  objectivec: 'Objective-C',
+};
+function MultiCodeBlock(props) {
+  return (
+    <Container>
+      {renderHeader(props.title, props.languages)}
+      {Array.isArray(props.children) ? props.children[0] : props.children}
+    </Container>
+  );
+
+  function renderHeader(title, languages = []) {
+    if (title || languages.length > 1) {
+      return (
+        <Header>
+          <HeaderInner>
+            <HeaderText>{props.title}</HeaderText>
+            <SpacingsInline
+              scale="m"
+              alignItems="center"
+              justifyContent="flex-end"
+            >
+              {renderLanguages(languages)}
+            </SpacingsInline>
+          </HeaderInner>
+        </Header>
+      );
+    }
+
+    return null;
+  }
+
+  function renderLanguages(languages) {
+    if (languages.length > 1) {
+      return (
+        <LanguagesDropDownWrapper>
+          <LanguagesDropDown onChange={handleOnLanguageChange}>
+            {languages.map(lang => (
+              <option key={lang} value={lang}>
+                {languageDisplayNames[lang] || lang}
+              </option>
+            ))}
+          </LanguagesDropDown>
+        </LanguagesDropDownWrapper>
+      );
+    }
+
+    return languages[0] === 'text' ? null : (
+      <HeaderText>
+        {languageDisplayNames[languages[0]] || languages[0]}
+      </HeaderText>
+    );
+  }
+
+  function handleOnLanguageChange(e) {
+    console.log(e.target.value);
+    // const newProps = props.codeBlockProps.find(
+    //   codeBlockProp => codeBlockProp.language === e.target.value
+    // );
+
+    // console.log(props);
+
+    // React.Children.map(props.children, (child, index) => {
+    //   console.log(props.children);
+    //   console.log(child);
+    //   // React.cloneElement(child, {
+    //   //   isActive: activeLanguageIndex === index,
+    //   // })
+    // });
+
+    // console.log(newProps);
+
+    // setCurrentCodeBlockProps(newProps);
+  }
+}
+
+MultiCodeBlock.propTypes = {
+  title: PropTypes.string,
+  languages: PropTypes.arrayOf(PropTypes.string.isRequired).isRequired,
```

Comment:
FYI: this prop is not necessary. The list of languages should be extracted from the children:

```js
const languages = React.Children.map(props.children, (child) => {
  const languageCode = child.props.language || 'text';
  return languageAliases[languageCode] || languageCode;
})
```